### DISCUSSION TITLE
Scatter plot: Programaticaly show all tooltips

### DISCUSSION THREAD

**Discussion by tomasonjo** [plouc/nivo#1667] [Q&A]
I would like to add a toggle button that shows/hide all tooltips in scatter plot programatically. There seems to be no setting to let me achieve this. I have seen that it is possible to add custom HTML over the scatter plot like so: https://nivo.rocks/storybook/?path=/story/scatterplot--synchronizing-charts

But I can't find any method to iterate over all the nodes in the visualization, so i could add a HTML component over all the data points in the visualization. Is there someway I could achieve this?

**Comment by nomanilyas16** [comment]
I think you can create your custom tooltip function and pass it to the tooltip prop. Add a toggle button and use it in the custom tool tip function as a condition like `if toggle === true {<div>Custom Tooltip</div>} else {}`.

**Comment by plouc** [comment]
The solution with a custom tooltip wouldn't work as only one will be displayed at a time. Such elements wouldn't be tooltips technically, to achieve something like this, you would have to create a custom layer, but I don't have a pre-made example available for this.
